framework/src/org/checkerframework/framework/util/AnnotatedTypes.java

```diff
@@ -1440,16 +1444,22 @@ public static boolean hasTypeQualifierElementTypes(ElementType[] elements) {
             } else if (!element.equals(ElementType.TYPE_PARAMETER)) {
                 // if there's an ElementType with a enumerated value of something other than
                 // TYPE_USE or TYPE_PARAMETER then it isn't a valid annotation
-                hasOtherElementTypes = true;
+                otherElementType = element;
+            }
+            if (hasTypeUse && otherElementType != null) {
+                checker.report(Result.warning("inconsistent.target.metaannotation", cls.getName(), otherElementType),
+                               // TODO: what should I pass as the source (an Element or Tree) here?
```

Comment:
No tree or element seems appropriate here.  Maybe we should add a version of SourceChecker#report that does not expect an element or tree?